open_tms_backend/open_tms_close_ticket/index.js

```diff
@@ -1,5 +1,176 @@
+ *               properties:
+ *                 meta:
+ *                   type: object
+ *                   properties:
+ *                     status:
+ *                       type: integer
+ *                       example: 400
+ *                     message:
+ *                       type: string
+ *                       example: Bad Request
+ *                 errors:
+ *                   type: array
+ *                   items:
+ *                     type: object
+ *                     properties:
+ *                       code:
+ *                         type: string
+ *                         example: BAD_REQUEST
+ *                       message:
+ *                         type: string
+ *                         example: Next stage or assignee is invalid.
+ *       500:
+ *         description: Internal Server Error
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 meta:
+ *                   type: object
+ *                   properties:
+ *                     status:
+ *                       type: integer
+ *                       example: 500
+ *                     message:
+ *                       type: string
+ *                       example: Something went wrong.
+ *                 errors:
+ *                   type: array
+ *                   items:
+ *                     type: object
+ *                     properties:
+ *                       code:
+ *                         type: string
+ *                         example: INTERNAL_ERROR
+ *                       message:
+ *                         type: string
+ *                         example: Something went wrong.
+ */
+const closeTicket = async (ticketId) => {
+  const TicketRevision = await shared.import('open_tms', 'TicketRevision')
+  const latestRevision = await TicketRevision.findOne({
+    where: { ticketId },
+    order: [['id', 'DESC']],
+  })
+
+  if (await validateProcessPath(latestRevision.currentStage, 'ticket_closed', latestRevision.assignee)) {
+    const Ticket = await shared.import('open_tms', 'Ticket')
+    const ticket = await Ticket.findByPk(ticketId)
+    ticket.status = 'closed'
+    await ticket.save()
+
+    // Create a new entry in the ticket_activity table with current_stage as "ticket_closed" id from stage table
+    const TicketActivity = await shared.import('open_tms', 'TicketActivity')
+    const activity = await TicketActivity.create({
+      TicketRevisionId: latestRevision.id,
+      currentStage: 'ticket_closed',
+    })
+
+    return { success: true, activity }
+  } else {
+    // If validation fails, send proper error message
+    throw new Error('Next stage or assignee is invalid.')
+  }
+}
+
+const validateProcessPath = async (currentStage, nextStage, assignee) => {
+  const ProcessPath = await shared.import('open_tms', 'ProcessPath')
+  const processPath = await ProcessPath.findOne({
```

Comment:
The process path requires from_stage (fetched from the latest ticket activity) and to_stage( id of the stage for which is_end flag is true in the stage table)